File: java/components/validation-framework/HEAD/validation-framework-test-utils/src/main/java/org/alljoyn/validation/framework/utils/introspection/bean/InterfaceDetail.java
/*******************************************************************************
 *   *  
 *     Copyright (c) 2016 Open Connectivity Foundation and AllJoyn Open
 *     Source Project Contributors and others.
 *     
 *     All rights reserved. This program and the accompanying materials are
 *     made available under the terms of the Apache License, Version 2.0
 *     which accompanies this distribution, and is available at
 *     http://www.apache.org/licenses/LICENSE-2.0

 *******************************************************************************/
package org.alljoyn.validation.framework.utils.introspection.bean;

import java.util.List;

public class InterfaceDetail
{
    private String path;
    private List<IntrospectionInterface> introspectionInterfaces;

    public InterfaceDetail(String path, List<IntrospectionInterface> introspectionInterfaces)
    {
        this.path = path;
        this.introspectionInterfaces = introspectionInterfaces;
    }

    public String getPath()
    {
        return path;
    }

    public List<IntrospectionInterface> getIntrospectionInterfaces()
    {
        return introspectionInterfaces;
    }
}

File: cpp/components/validation-ctt/HEAD/ctt_testcases/InterfaceDetail.cpp
/******************************************************************************
* * 
*    Copyright (c) 2016 Open Connectivity Foundation and AllJoyn Open
*    Source Project Contributors and others.
*    
*    All rights reserved. This program and the accompanying materials are
*    made available under the terms of the Apache License, Version 2.0
*    which accompanies this distribution, and is available at
*    http://www.apache.org/licenses/LICENSE-2.0

******************************************************************************/
#include "stdafx.h"
#include "InterfaceDetail.h"

using namespace std;

InterfaceDetail::InterfaceDetail(string t_Path, list<IntrospectionInterface> t_IntrospectionInterfaces) :
m_Path(t_Path), m_IntrospectionInterfaces(t_IntrospectionInterfaces) {}

string InterfaceDetail::getPath()
{
	return m_Path;
}

list<IntrospectionInterface> InterfaceDetail::getIntrospectionInterfaces()
{
	return m_IntrospectionInterfaces;
}